### PR TITLE
Performance tests

### PR DESCRIPTION
**Pull request by tronxd**
- Use `pytest-benchmark` plugin to evaluate execution times of different IaC runners on known benchmarks.

- Performance tests will fail if they deviate by a configurable percecnt from the maximum duration of the tested branch against the data. 
- Thresholds are set according to current execution on the data.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -85,3 +85,43 @@ jobs:
       - name: Run integration tests
         run: |
           pipenv run pytest integration_tests -k 'not api_key'
+
+  performance-tests:
+    strategy:
+      fail-fast: true
+      matrix:
+        python: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (nimrodkor) [inline .github/workflows/pr-test.yml:93]: @tronxd Won't the runtime be a little different in each python version? What does running it on all python runtimes give us?

I'd expect we would run only on 1 runtime, doesn't matter which one IMO

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -85,3 +85,43 @@ jobs:
...
+        with:
+          # version 2021.11.5, 2021.11.5.post0 do not work
+          version: 2021.5.29
+      - name: Build & install checkov package
+        run: |
+          pipenv --python ${{ matrix.python }}
+          pipenv run pip install --upgrade pip==21.1.1
+          pipenv run pip install pytest pytest-benchmark
+          pipenv run python setup.py sdist bdist_wheel
+          bash -c 'pipenv run pip install dist/checkov-*.whl'
```
Comment (nimrodkor) [inline .github/workflows/pr-test.yml:114]: @tronxd I checked the tests out - we're not using the packaged checkov. So why do we need to build & install checkov?

File: .github/workflows/pr-test.yml [hunk 3]
```diff
@@ -85,3 +85,43 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/setup-node@v2
+      - uses: azure/setup-helm@v1
+      - uses: dschep/install-pipenv-action@v1
+        with:
+          # version 2021.11.5, 2021.11.5.post0 do not work
```
Comment (nimrodkor) [inline .github/workflows/pr-test.yml:106]: @tronxd Why does it not work? How old is the version we're using?

**Review by nimrodkor — COMMENTED** [review]
@tronxd Nice!!

File: .github/workflows/pr-test.yml (hunk 3 above)
Comment (gruebel) [reply .github/workflows/pr-test.yml:106]: there was an issue with both of the version, I pinned it on the weekend. They published a new one, probably worth a try. Remove `with` block -> run it -> works then great, if not put it back

File: .github/workflows/pr-test.yml (hunk 2 above)
Comment (tronxd) [reply .github/workflows/pr-test.yml:114]: @nimrodkor The tests measure different checkov runners duration. so checkov is needed

File: .github/workflows/pr-test.yml (hunk 3 above)
Comment (tronxd) [reply .github/workflows/pr-test.yml]: @gruebel Worked, thanks!

File: .github/workflows/pr-test.yml (hunk 2 above)
Comment (nimrodkor) [reply .github/workflows/pr-test.yml:111]: Checkov the code exists - why do we need the packaged version?
Are the tests bashing out and running `checkov ...`? I didn't see that

File: .github/workflows/pr-test.yml [hunk 4]
```diff
@@ -85,3 +85,40 @@ jobs:
...
+      matrix:
+        python: ["3.7"]
+    env:
+      working-directory: ./performance_tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
```
Comment (nimrodkor) [inline .github/workflows/pr-test.yml:101]:
```suggestion
  performance-tests:
    env:
      working-directory: ./performance_tests
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: 3.7
```

**Pull request merged**